### PR TITLE
feat: [AXIMST-469] Unit page - display unit and xblock visibility messages

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** Unit page - display unit and xblock visibility messages

**Youtrack:** https://youtrack.raccoongang.com/issue/AXIMST-469

### CODE REVIEW

File: src/course-unit/course-xblock/CourseXBlock.test.jsx [hunk 1]
```diff
@@ -290,4 +290,22 @@ describe('<CourseXBlock />', () => {
...
+      {
+        userPartitionInfo: {
+          ...camelCaseObject(userPartitionInfo),
+          selectedGroupsLabel: 'Visibility group 1',
+        },
+      },
+    );
+
+    await waitFor(() => {
+      // userEvent.click(duplicateBtn);
```
Comment (ruzniaievdm) [inline src/course-unit/course-xblock/CourseXBlock.test.jsx:305]: Please remove it

File: src/course-unit/course-xblock/CourseXBlock.test.jsx [hunk 2]
```diff
@@ -290,4 +290,22 @@ describe('<CourseXBlock />', () => {
       expect(mockedUsedNavigate).toHaveBeenCalledWith(`/course/${courseId}/editor/problem/${id}`);
     });
   });
+
+  it('displays a visibility message if item has accessible restrictions', async () => {
+    const { getByText } = renderComponent(
+      {
+        userPartitionInfo: {
+          ...camelCaseObject(userPartitionInfo),
```
Comment (ruzniaievdm) [inline src/course-unit/course-xblock/CourseXBlock.test.jsx:298]: [important] It looks like we can convert each keys into camelCase in `courseVerticalChildrenMock` file instead using camelCaseObject every time.

File: src/course-unit/course-xblock/xblock-messages/XBlockMessages.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const intl = useIntl();
+  const type = getMessagesBlockType(validationMessages);
+
+  if (!validationMessages.length) {
+    return null;
+  }
+
+  return (
+    <Alert
+      variant={type === MESSAGE_ERROR_TYPES.warning ? 'warning' : 'danger'}
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/XBlockMessages.jsx:20]: [question]: Can we put this code into a variable?
```suggestion
     const { warning } = MESSAGE_ERROR_TYPES;
     const alertVariant = (type === warning) ? 'warning' : 'danger';
     ...
     variant={alertVariant}
```

File: src/course-unit/course-xblock/xblock-messages/XBlockMessages.jsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      icon={type === MESSAGE_ERROR_TYPES.warning ? WarningIcon : InfoIcon}
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/XBlockMessages.jsx:21]:
```suggestion
      const iconVariant = (type === warning) ? WarningIcon : InfoIcon
      ...
      icon={iconVariant}
```

File: src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import XBlockMessages from './XBlockMessages';
+import messages from '../messages';
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx:6]:
```suggestion
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import messages from '../messages';
import XBlockMessages from './XBlockMessages';
```

File: src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+const renderComponent = (props) => render(
+  <IntlProvider locale="en">
+    <XBlockMessages
+      {...props}
+    />
+  </IntlProvider>,
+);
+
+describe('XBlockMessages Component', () => {
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx:16]: [optional]:
```suggestion
describe('<XBlockMessages />', () => {
```

File: src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('does not render anything when there are no errors', () => {
+    const { container } = renderComponent({ validationMessages: [] });
+    expect(container.firstChild).toBeNull();
+  });
+
+  it('renders a warning Alert when there are warning errors', () => {
+    const validationMessages = [{ type: 'warning', text: 'This is a warning' }];
+    const { getByText } = renderComponent({ validationMessages });
+    expect(getByText('This is a warning')).toBeInTheDocument();
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx:29]:
```suggestion
    const validationMessages = [{ type: 'warning', text: 'This is a warning' }];
    const { getByText } = renderComponent({ validationMessages });

    expect(getByText('This is a warning')).toBeInTheDocument();
```

File: src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    expect(getByText(messages.validationSummary.defaultMessage)).toBeInTheDocument();
+  });
+
+  it('renders a danger Alert when there are danger errors', () => {
+    const validationMessages = [{ type: 'danger', text: 'This is a danger' }];
+    const { getByText } = renderComponent({ validationMessages });
+    expect(getByText('This is a danger')).toBeInTheDocument();
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/XBlockMessages.test.jsx:36]:
```suggestion
    const validationMessages = [{ type: 'danger', text: 'This is a danger' }];
    const { getByText } = renderComponent({ validationMessages });

    expect(getByText('This is a danger')).toBeInTheDocument();
```

File: src/course-unit/course-xblock/xblock-messages/utils.test.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import { MESSAGE_ERROR_TYPES } from '../constants';
+import { getMessagesBlockType } from './utils';
+
+describe('xblock-messages utils', () => {
+  describe('getMessagesBlockType', () => {
+    test('returns "warning" when there are no error messages', () => {
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/utils.test.js:6]: [nit]: please replace "test" with "it" in this file
```suggestion
    test('returns "warning" when there are no error messages', () => {
```

File: src/course-unit/course-xblock/xblock-messages/utils.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import { MESSAGE_ERROR_TYPES } from '../constants';
+
+// eslint-disable-next-line import/prefer-default-export
+export const getMessagesBlockType = (messages) => {
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-messages/utils.js:4]: [nit]: let's write jsDocs for this function

File: src/course-unit/course-xblock/xblock-content/XBlockContent.test.jsx [hunk 11]
```diff
@@ -30,12 +30,12 @@ const props = {
...
 const RootComponent = () => (
   <IntlProvider locale="en">
-    <ContentIFrame {...props} />
+    <XBlockContent {...props} />
   </IntlProvider>
 );
 
 let el;
-describe('ContentIFrame Component', () => {
+describe('XBlockContent Component', () => {
```
Comment (PKulkoRaccoonGang) [inline src/course-unit/course-xblock/xblock-content/XBlockContent.test.jsx:38]: [optional]:
```suggestion
describe('<XBlockContent />', () => {
```

**Review by PKulkoRaccoonGang — CHANGES_REQUESTED** [review]

**Review by monteri — APPROVED** [review]

**Pull request merged**